bokeh/core/validation/check.py

```diff
@@ -125,6 +125,9 @@ def check_integrity(models):
         if code not in __silencers__:
             log.warning("W-%d (%s): %s: %s" % msg)
 
+    # QUESTION: can I make check_integrity return messages?
+    return messages
```

Comment:
This should not be necessary, message need to be handled entirely inside `check_integrity`. The commented code below should already almost be exactly what is needed, I would only expect it to need to be updated to handle the new setting value that replaces `strict`